tests/navier_stokes/stokes_operator.cpp

```diff
@@ -158,34 +226,67 @@ main(int argc, char* argv[])
         const double C = input_db->getDouble("C");
         poisson_spec.setDConstant(D);
         poisson_spec.setCConstant(C);
+        if(periodic_shift.min() > 0)
+        {
+            StaggeredStokesOperator stokes_op("stokes_op", true);
 
-        // Setup the stokes operator
-        StaggeredStokesOperator stokes_op("stokes_op", true);
-
-        stokes_op.setVelocityPoissonSpecifications(poisson_spec);
-        stokes_op.initializeOperatorState(u_vec, f_vec);
+            stokes_op.setVelocityPoissonSpecifications(poisson_spec);
+            stokes_op.initializeOperatorState(u_vec, f_vec);
+            // Apply the operator
+            stokes_op.apply(u_vec, f_vec);
+        }
+        else
+        {
+            Pointer<StaggeredStokesPhysicalBoundaryHelper> bc_helper = new StaggeredStokesPhysicalBoundaryHelper;
+            bc_helper->cacheBcCoefData(u_bc_coefs,0.0,patch_hierarchy);
+            bc_helper->copyDataAtDirichletBoundaries(e_sc_idx, u_sc_idx);
+            // Setup the stokes operator
+            StaggeredStokesOperator stokes_op("stokes_op", false, input_db);
+            stokes_op.setPhysicalBcCoefs(u_bc_coefs,NULL);
+            stokes_op.setVelocityPoissonSpecifications(poisson_spec);
+            stokes_op.setPhysicalBoundaryHelper(bc_helper);
+            stokes_op.initializeOperatorState(u_vec, f_vec);
 
-        // Apply the operator
-        stokes_op.apply(u_vec, f_vec);
+            // Apply the operator
+            stokes_op.apply(u_vec, f_vec);
+        }
 
         // Compute error and print error norms.
-        e_vec.subtract(Pointer<SAMRAIVectorReal<NDIM, double> >(&f_vec, false),
-                       Pointer<SAMRAIVectorReal<NDIM, double> >(&e_vec, false));
+        e_vec.subtract(Pointer<SAMRAIVectorReal<NDIM, double>>(&f_vec, false),
+                       Pointer<SAMRAIVectorReal<NDIM, double>>(&e_vec, false));
+        //print out the errors in each norm
         pout << "|e|_oo = " << e_vec.maxNorm() << "\n";
         pout << "|e|_2  = " << e_vec.L2Norm() << "\n";
         pout << "|e|_1  = " << e_vec.L1Norm() << "\n";
+        
+        
+        
+
+        // Interpolate the side-centered data to cell centers for output.
+        static const bool synch_cf_interface = true;
+        hier_math_ops.interp(draw_u_idx, draw_u_var, u_sc_idx, u_sc_var, nullptr, 0.0, synch_cf_interface);
+        hier_math_ops.interp(draw_f_idx, draw_f_var, f_sc_idx, f_sc_var, nullptr, 0.0, synch_cf_interface);
+        hier_math_ops.interp(draw_e_idx, draw_e_var, e_sc_idx, e_sc_var, nullptr, 0.0, synch_cf_interface);
+
+        // Output data for plotting.
+        visit_data_writer->writePlotData(patch_hierarchy, 0, 0.0);
 
         // Deallocate level data
         // Allocate data on each level of the patch hierarchy.
+        
+        //Question-> Why does data need to be deallocated?
```

Comment:
```suggestion
```
We deallocate data to prevent memory leaks. While it may not be a big deal in this test (since data will be deallocated  in a few lines automatically), it's a good habit to get into.